### PR TITLE
chore(ui-select): add TS types to `ui-select` package

### PR DESCRIPTION
**Pull request by ke1k0**
Closes: INSTUI-3078

### CODE REVIEW

File: packages/ui-select/src/Select/index.tsx [hunk 1]
```diff
@@ -126,19 +119,20 @@ class Select extends Component<SelectProps> {
     hasInputRef: false
   }
 
-  _defaultId = uid('Select')
-  _assistId = uid('Select-assistiveText')
```
Comment (ke1k0) [inline packages/ui-select/src/Select/index.tsx:130]: this wasn't used anywhere

File: packages/ui-select/src/Select/index.tsx [hunk 2]
```diff
@@ -458,53 +442,50 @@ class Select extends Component<SelectProps> {
...
           {...getListProps({ as: 'ul', elementRef: this.handleListRef })}
         >
           {isShowingOptions
-            ? // @ts-expect-error ts-migrate(7030) FIXME: Not all code paths return a value.
-              Children.map(children, (child, index) => {
+            ? Children.map(children, (child, index) => {
                 if (!child || !matchComponentTypes(child, [Group, Option])) {
                   return // ignore invalid children
                 }
                 if (matchComponentTypes(child, [Option])) {
```
Comment (matyasf) [inline packages/ui-select/src/Select/index.tsx:478]: We should modernize this code in the future, there is lots of boilerplate going on just for these comparisons (if I understand correctly, this code is the sole reason for Select.Group and Select.Option's existence)

**Review by matyasf — APPROVED** [review]
looks good!

**Comment by matyasf** [comment]
There are lots of change is Chromatic, do you know why?

**Review by Brailor — APPROVED** [review]

**Comment by ke1k0** [comment]
> There are lots of change is Chromatic, do you know why?

@matyasf I checked and the number of examples are the same on master and here, but since I shuffled the props around a lot, I assume only the order of the examples changed.

File: packages/ui-select/src/Select/index.tsx (hunk 2 above)
Comment (ke1k0) [reply packages/ui-select/src/Select/index.tsx:478]: It is indeed pretty weird, but we already started the talks about writing a new Select component and retire this one...

**Pull request merged**